applications/rubintv/templates/deployment-frontend.yaml

```diff
@@ -0,0 +1,102 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ template "rubintv.fullname" . }}-frontend
+  labels:
+    {{- include "rubintv.labels" . | nindent 4 }}
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      {{- include "rubintv.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      {{- with .Values.frontend.podAnnotations }}
+      annotations:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      labels:
+        {{- include "rubintv.selectorLabels" . | nindent 8 }}
+    spec:
+      automountServiceAccountToken: false
+      {{- with .Values.workers.imagePullSecrets }}
+      imagePullSecrets:
+      {{ toYaml . | nindent 8 }}
+      {{- end }}
+      containers:
+        - name: frontend
+          env:
+            - name: "ENVIRONMENT_URL"
+              value: {{ .Values.global.baseUrl | quote }}
+            - name: "SAFIR_PATH_PREFIX"
+              value: {{ .Values.frontend.pathPrefix | quote }}
+            - name: "AWS_SHARED_CREDENTIALS_FILE"
+              value: /etc/secrets/aws-credentials.ini
+            {{- if (not .Values.frontend.debug) }}
+            - name: "SAFIR_PROFILE"
+              value: "production"
+            {{- end }}
+            {{- with .Values.frontend.env }}
+            {{- toYaml . | nindent 12 }}
+            {{- end }}
+          image: "{{ .Values.frontend.image.repository }}:{{ .Values.frontend.image.tag | default .Chart.AppVersion }}"
+          imagePullPolicy: {{ .Values.frontend.image.pullPolicy | quote }}
+          ports:
+            - name: "http"
+              containerPort: 8080
+              protocol: "TCP"
+          readinessProbe:
+            httpGet:
+              path: "/"
+              port: "http"
+            timeoutSeconds: 10
+          {{- with .Values.frontend.resources }}
+          resources:
+            {{- toYaml . | nindent 12 }}
+          {{- end }}
+          securityContext:
+            allowPrivilegeEscalation: false
+            capabilities:
+              drop:
+                - ALL
+            readOnlyRootFilesystem: true
+          volumeMounts:
+            - name: "user-secrets"
+              mountPath: "/etc/secrets"
+      securityContext:
+        runAsNonRoot: true
+        runAsUser: 1000
+        runAsGroup: 1000
+      initContainers:
+        - name: "secret-perm-fixer-frontend"
+          image: "busybox"
+          command:
+            - "/bin/sh"
+            - "-c"
+            - |
+              cp /secrets/* /etc/secrets && \
+              chown 1000:1000 /etc/secrets/* && \
+              chmod 0600 /etc/secrets/*
```

Comment:
If there's no postgres secret, you don't need to do any of this. This is only required for postgres, and I think you said elsewhere that the frontend doesn't need postgres credentials? If that's true, you can delete all of this; the default mode for mounted secrets is 0644, and only the postgres libraries care about the ownership.

Reply:
I'll take it out and check that the S3 creds are still readable.  I feel like I started there and something failed, but I might be misremembering.

Reply:
Works fine.